### DISCUSSION TITLE
First download docker images, second start containers

### DISCUSSION THREAD

**Discussion by daysus** [engageub/InternetIncome#96] [Ideas]
It is more stable for the server to first download all the docker images of the monetizers that have been configured and second to start the different containers.

Internally, the connections are combined with the discharges and it becomes congested.

**Comment by engageub** [comment]
Hello,
Docker images are only downloaded once on the host.
From the next time, if you run the script, it will check if there is a latest version and download if there is any update. 
Even if you run 1000 proxies, it will only be downloaded once and only for the first proxy. 
Please note that the script is single threaded and executes serially after the completion of previous step. 

Thank you

**Reply by ppzzAn** [reply]
Hello

After new update with randomised names, not connecting to honeygain. They have saved old hardware names, and logs says you have exceeded limits with connected devices.

Any suggestions what to do with it ?

Also ebesucher link doesn't open firefox docker image. Chrome works fine
![Screenshot_54](https://github.com/engageub/InternetIncome/assets/129802150/1e73ef15-7642-459f-b718-8a4df12271a0)

**Reply by engageub** [reply]
Hello,
If you are trying to delete and create the containers, new device names are created in test branch.
Honeygain has a limit of 10 devices. You either have to wait for sometime for the old device data to be deleted or use other account or use the main branch to set the device names.
If its taking more time to delete the data, option will be provided to manually set the device names as it was earlier.

Regarding ebesucher, could you please let me know if you are using serveo.net or direct link to access it.
Is the screenshot attached, related to firefox inside the container or outside the container.


Thank you

**Reply by engageub** [reply]
Please look into the logs of ebesucher for any potential issues.
It can also be related to the recent update of firefox.
https://hub.docker.com/r/jlesage/firefox
You may use main branch till then. 

Thank you

**Reply by engageub** [reply]
For ebesucher, it  requires a minimum resolution of1280 x 1024. You will have to set EBESUCHER_USE_RANDOM_DISPLAY=true
Please see the discussion here https://github.com/engageub/InternetIncome/discussions/79

Thank you

**Reply by ppzzAn** [reply]
Yes, Thank you

[xvnc        ] (EE) 
[xvnc        ] Fatal server error:
[xvnc        ] (EE) Invalid geometry x passed to -geometry
[xvnc        ] (EE) 

Also speedshare does it works? seems no any data exchange happening

**Reply by ppzzAn** [reply]
Random_Display fixed the issue. Great, Thank you.
High possibility for HoneyGain need to wait 14 days before device list updated. As It not connect under new name.

**Reply by engageub** [reply]
Code has been updated now and random device generation in test branch has been removed due to honeygain issues.

**Reply by engageub** [reply]
Speedshare is working but it took about 2 days for the traffic to show up on the devices in dashboard.

**Reply by engageub** [reply]
Also note that ebesucher and adnade do not work together when chrome is used in test branch.

**Reply by ppzzAn** [reply]
Now ebesucher window stay blank with those message
![Screenshot_55](https://github.com/engageub/InternetIncome/assets/129802150/7dc6b552-5161-451c-b43d-789d590f33c3)

**Comment by engageub** [comment]
Yes, it appears and moves to next page.
Is it not moving to next page?

**Reply by ppzzAn** [reply]
No just stuck, reload not working aswel. Only one browser from 10

**Reply by engageub** [reply]
I don't see any issue here. It shows the IP address and waits for about 30 seconds and moves to next page. Please make sure you have sufficient CPU and memory. Or you can try increasing the CPU from 1 to 2 if your CPU frequency is low.
There can also be an issue with your proxy.

Thank you

**Reply by ppzzAn** [reply]
Ok, Thank you.

**Reply by ppzzAn** [reply]
Hello

Any ideas why ebesucher in firefox crashing ?
![Screenshot_56](https://github.com/engageub/InternetIncome/assets/129802150/837553e2-618e-4651-bbc3-bcf8a25aca29)

**Reply by engageub** [reply]
Hi,
Is this happening in main branch or test branch?
You may try main branch and update if there is any issue.
There are many reasons why that may appear.
It restarts every two hours in test branch and every one hour in main branch. 
Please make sure you have sufficient cpu and memory also. 

Thank you

**Reply by ppzzAn** [reply]
In test branch, I have 16gb ram and 6cores, only 10 browsers run.
I will try main one.

**Reply by engageub** [reply]
With 6 cores, could you please check your available CPU. In test branch, by default 1 CPU is assigned to each Firefox container. That's the maximum it consumes.
You can set the configuration in properties.conf
Thank you

**Reply by ppzzAn** [reply]
My settings and consumption
![Screenshot_58](https://github.com/engageub/InternetIncome/assets/129802150/3470b553-6ac7-4ed0-9233-ca17db0fb900)
![Screenshot_57](https://github.com/engageub/InternetIncome/assets/129802150/f5572fc6-47e1-4f6d-a64e-5987d3e67233)

**Reply by engageub** [reply]
Hello,
With 0.5 CPU, it may not work as expected. Please set to atleast 1 and try. The load will be distributed. 
Main branch does not have this restriction.

Thank you

**Reply by ppzzAn** [reply]
Ok

Always crashing on particular web page.
![Screenshot_57](https://github.com/engageub/InternetIncome/assets/129802150/ad7efc11-93ec-4358-b2df-68e4ad574599)
![Screenshot_58](https://github.com/engageub/InternetIncome/assets/129802150/969e8ca4-5cdb-497a-b587-f35ee0eba853)

**Comment by engageub** [comment]
Is this happening on main branch also?

Thank you

**Reply by ppzzAn** [reply]
I haven't tested main one yet. I will update later.

**Reply by engageub** [reply]
Thank you. Please update if this issue is present in main branch.

**Comment by engageub** [comment]
With limited cpu, multiple issues can occur when resources are not
available.
Thank you

**Reply by ppzzAn** [reply]
I have updated cpu settings and still have plenty resourses left.

**Reply by engageub** [reply]
May I know if you are using the default settings in test branch or have changed something while running it.

**Reply by ppzzAn** [reply]
I changed settings 

It looks like that now
MAX_MEMORY=1024m
MEMORY_RESERVATION=756m
CPU=1

**Reply by ppzzAn** [reply]
Time to time I have this problem occur
![Screenshot_58](https://github.com/engageub/InternetIncome/assets/129802150/e76e94ac-8fd9-435e-aa8e-9a1a5a0ff81b)

**Reply by ppzzAn** [reply]
Only one page crashing 
![Screenshot_59](https://github.com/engageub/InternetIncome/assets/129802150/5e2323e6-b3ee-41d7-a260-cdb77891e67a)

Maybe it trying download something, or script on page

**Reply by engageub** [reply]
Is this happening in main branch also? You have not confirmed me on that.
Also, in test branch is this happening in 1 hour or after 1 hour. 
I have been looking at it for 5 minutes and it works fine but I have not come across the website you mentioned.
Thank you

**Reply by engageub** [reply]
I tried to open it with 0.25 CPU, it loads slowly but did not crash.
The website is not loading as other websites. This has to be informed to ebesucher support team.
It is in loading state for more than a minute. 
Also, Hardware acceleration is already disabled in firefox. 
I will keep it running with ebesucher for sometime and see if it crashes here.
I am using arm64 architecture. 
You have the option to use chrome browser also in test branch. 
In addition to this I am still awaiting confirmation on main branch. 

![image](https://github.com/engageub/InternetIncome/assets/103608400/2211a8f6-361a-49d6-a8a5-0d7b387029cf)


Thank you

**Reply by engageub** [reply]
The urls usually have ebesucher.com in the navigation bar. 
I have been running over an hour and have not faced any issues.
The url you mentioned may not have appeared. However you may contact ebesucher team about the issue as firefox browser is being used.
I am also awaiting reply from you regarding the main branch.

**Reply by ppzzAn** [reply]
Hi

I think you get crash when dns resolving enabled for ebesucher.

Testing now, will update.

**Reply by engageub** [reply]
DNS has been changed to google now. Earlier it was cloudflare. May be cloudflare has blocked that request. 

Thank you

**Reply by ppzzAn** [reply]
Only for ebesucher ?

**Reply by engageub** [reply]
For all the containers google dns is used in the latest update due to cloudflare blocking the IP addresses. 

Thank you

**Reply by engageub** [reply]
To confirm if that's a DNS issue on the VPS, I would suggest you to run the following. If you are using socks5 DNS you have to check for dns issues on that ip address.
Also, you can try opening the url manually in firefox browser inside the container and see if it crashes.

```
nslookup forofosdelrunning.com 1.1.1.1
nslookup forofosdelrunning.com 1.0.0.1

```

Thank you

**Reply by ppzzAn** [reply]
Same got crash. I did run separed folder for ebesucher only without socks5 dns resolving, only in that way it was stable.

**Reply by ppzzAn** [reply]
Maybe UDP direct traffic crashes it. It just an idea

**Reply by engageub** [reply]
So, you are saying that socks5 DNS is causing issue and not with direct connection dns.

**Reply by ppzzAn** [reply]
Seems like, I will run more time in separate folder. But I can say now I haven't seen any crashes yet.

**Comment by ppzzAn** [comment]
Some webpages have video players and probably that's the root of the problem when DNS resolving enabled.

**Reply by engageub** [reply]
There are multiple reasons why it can happen. It depends on the design of firefox which I do not own.
May be there are threads/processes waiting for response and retrying and crashes. 
It cannot be concluded unless the code is checked and verified. Anything else will be assumptions. 
Of course issue can be replicated by trying different ways to narrow down the reason. But the actual root cause should be pointed out from the code point of view. 

Thank you

**Reply by ppzzAn** [reply]
I'm just trying to run few scenarios, to make sure it's DNS resolving problem. And not caused by CPU and Ram Limitation.

Thank you

**Reply by engageub** [reply]
At the moment google dns is being used. Earlier cloudflare dns was used. You may change that code in internetIncome.sh file.

**Reply by ppzzAn** [reply]
Done it.

**Reply by ppzzAn** [reply]
Crashing is not related with socks5 DNS, just lack of resources

**Reply by engageub** [reply]
How much have you set the cpu. Did you not use the default CPU configuration.
How were you checking the available CPU on the host.


Thank you

**Reply by ppzzAn** [reply]
I set 1 cpu, checking by htop command line.

**Reply by ppzzAn** [reply]
Anything less than 2064m and 1 cpu crash browser in VM

**Reply by engageub** [reply]
okay. Does it crash after 1 hour as the restart time is set to 2 hours in test branch.
Main branch does not have any configuration for cpu or memory. You have not confirmed to me anything about it when asked multiple times. That would narrow down the underlying issue. 
You can check the docker stats for firefox container to confirm on the memory being consumed when running. 

```
sudo docker stats
```

**Reply by engageub** [reply]
If you are using top command check the idle percentage. It will have different % for user and system. 
So you have to subtract 100 from idle percentage or %id to get the total consumption of cpu.

**Comment by ppzzAn** [comment]
> okay. Does it crash after 1 hour as the restart time is set to 2 hours in test branch.
> Main branch does not have any configuration for cpu or memory. You have not confirmed to me anything about it when asked multiple times. That would narrow down the underlying issue.
> You can check the docker stats for firefox container to confirm on the memory being consumed when running.

No while running, random time. I think I found right settings. Main branch not suitable for me as 10 browsers consuming everything from VPS and if I run more containers it just not enough memory.

**Reply by engageub** [reply]
okay. But its requires more than 2 GB RAM for each instance on your PC.
In main branch, when I used earlier it was consuming less than 1 GB of RAM.
May be recent changes in firefox increased it or there is something else that's causing it on your pc.
However, it restarts automatically after 2 hours with new firefox instance.

Thank you

**Reply by engageub** [reply]
With the recent update, it restarts every one hour. So memory(cache) consumed would be usually less when compared to what it was earlier with 2 hour restart. However it cannot be guaranteed that it will remain the same as webpages with large media files, complex scripts, and extensive content can require more memory/cpu to render and display.

**Reply by engageub** [reply]
I tried to limit the memory to 256m but the crash message did not appear on ebesucher firefox container. However, I was able to replicate in firefox in windows desktop by running the following script.
It did not happen all the time. It can be due to CPU or memory or network.
Please find below the firefox documentation. You may submit the report to firefox regarding this, 

https://support.mozilla.org/en-US/kb/troubleshoot-firefox-crashes-closing-or-quitting


```
while (true) {
  try {
    new ArrayBuffer(1024 * 1024 * 1024); // Attempt to allocate 1 GB ArrayBuffer
  } catch (e) {
    console.error("Memory allocation failed:", e);
    break;
  }
}
```

![image](https://github.com/engageub/InternetIncome/assets/103608400/6c141a7b-645c-40ad-b3f8-66692d1d2994)

**Comment by ppzzAn** [comment]
> With the recent update, it restarts every one hour. So memory(cache) consumed would be usually less when compared to what it was earlier with 2 hour restart. However it cannot be guaranteed that it will remain the same as webpages with large media files, complex scripts, and extensive content can require more memory/cpu to render and display.

Seems like it related with new update of firefox. Before when I did run the script that never happened. Even settings was much lower.
I've got fresh update.
Thank you.

**Comment by engageub** [comment]
Yes, firefox was updated 8 days ago. In that case, to use the old firefox, you will have to rename jlesage/firefox
 to jlesage/firefox:v23.11.3 in internetIncome.sh

Thank you

**Reply by ppzzAn** [reply]
I will try older version. 

Thank you
